pennylane_lightning/src/StateVector.hpp

```diff
@@ -15,19 +15,717 @@
+     */
+    void applyOperations(const vector<string> &ops,
+                         const vector<vector<size_t>> &wires,
+                         const vector<bool> &inverse,
+                         const vector<vector<fp_t>> &params = {{}}) {
+        const size_t numOperations = ops.size();
+        if (numOperations != wires.size() || numOperations != params.size())
+            throw std::invalid_argument(
+                "Invalid arguments: number of operations, wires, and "
+                "parameters must all be equal");
+
+        for (size_t i = 0; i < numOperations; i++) {
+            applyOperation(ops[i], wires[i], inverse[i], params[i]);
+        }
+    }
+
+    static constexpr vector<CFP_t> getPauliX() {
+        return {ZERO, ONE, ONE, ZERO};
+    }
+    static constexpr vector<CFP_t> getPauliY() {
+        return {ZERO, -IMAG, IMAG, ZERO};
+    }
+    static constexpr vector<CFP_t> getPauliZ() {
+        return {ONE, ZERO, ZERO, -ONE};
+    }
+    static constexpr vector<CFP_t> getHadamard() {
+        return {INVSQRT2, INVSQRT2, INVSQRT2, -INVSQRT2};
+    }
+    static constexpr vector<CFP_t> getS() { return {ONE, ZERO, ZERO, IMAG}; }
+    static constexpr vector<CFP_t> getT() { return {ONE, ZERO, ZERO, IMAG}; }
+    static constexpr vector<CFP_t> getCNOT() {
+        return {ONE,  ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO,
+                ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ONE,  ZERO};
+    }
+    static constexpr vector<CFP_t> getSWAP() {
+        return {ONE,  ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO,
+                ZERO, ONE,  ZERO, ZERO, ZERO, ZERO, ZERO, ONE};
+    }
+    static constexpr vector<CFP_t> getCZ() {
+        return {ONE,  ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO,
+                ZERO, ZERO, ONE,  ZERO, ZERO, ZERO, ZERO, -ONE};
+    }
+
+    static constexpr vector<CFP_t> getCSWAP() {
+        return {ONE,  ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO,
+                ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO, ZERO, ZERO, ZERO,
+                ZERO, ZERO, ZERO, ONE};
+    }
+    static constexpr vector<CFP_t> getToffoli() {
+        return {ONE,  ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO, ZERO,
+                ZERO, ZERO, ZERO, ZERO, ZERO, ONE,  ZERO, ZERO, ZERO, ZERO,
+                ZERO, ZERO, ONE,  ZERO};
+    }
```

Comment:
Yes. These were all defined for each gate class so I implemented them for brevity. Since we do not use them here though it may be fine to move them to a separate header/class. What do you think?

Reply:
Could we just remove them entirely (unless we expect to use in future?)

Reply:
Yep, pulling them out of here seems fine. I use some of these for the tests, and I think it will be good practice to allow pulling the gate definition out as needed, so will likely just move these to their own namespace `Pennylane::Gates` and allow them to be called from there.